### ISSUE TITLE
Decorator can't act on Tags that are not at root.

### ISSUE THREAD

**Issue by sammachin** [Redocly/redocly-cli#1028]
**Describe the bug**
Our swagger spec has tags as part of the operation, I am trying to write a custom decorator to reformat these, however my code will not trigger on the TagList item if the tags are not at the root level.

**To Reproduce**
This is a sample of the OAS spec
```
{
  "openapi": "3.0.1",
  "info": {
    "title": "Test",
    "version": "1.0"
  },
  "tags": [
    "TestTag"
  ],
  "paths": {
    "/api/v1/apitokens": {
      "get": {
        "summary": "List the API Tokens used to access this API",
        "tags": [
          "ApiTokensControllerV1"
        ],
        "responses": {
          "200": {
            "description": "Success",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ApiTokenReadModelV1ListMethodResult"
                }
              },
            }
          }
        }
      }
    }
  }
}
```
And this is my custom plugin
```
module.exports = RewriteTags;

/** @type {import('@redocly/cli').OasDecorator} */
function RewriteTags({tag}) {
  return {
    TagList: {
      leave(t) {
        console.log(t)
      }
    }
  }
};
```

When this is run it logs out the `TestTag` at the root but not the `ApiTokensControllerV1` tag in the get operation

**Expected behavior**
Should act on tags at all levels

**Logs**

**OpenAPI definition**
See above

**Redocly Version(s)**
1.0.0-beta.124

**`Node.js` Version(s)**
v16.13.0

**Comment by lornajane** [comment]
Thanks @sammachin ! I think this is a good suggestion, but that we might need a different object to represent the tags in the operations (`OperationTag`) because they're just an array of strings. The top-level `tags` element should define name and description for each tag, then just the key of that structure is used in the per-operation array. `OperationTagList` could be a useful addition so you can change what's in the list of each operation, depending on your use case.

**Comment by tatomyr** [comment]
Hi @sammachin, 

As @lornajane already mentioned, your root-level tags should be different from the operation-level tags and they are invalid according to [the spec](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#tag-object). You can run `redocly lint` command to see the errors.
 
However, you can still reference operation-level tags in your custom decorator like this (see more details in [the docs](https://redocly.com/docs/cli/resources/custom-plugins/)):

```ts
module.exports = RewriteTags;

/** @type {import('@redocly/cli').OasDecorator} */
function RewriteTags({tag}) {
  return {
    Operation: {
      leave(o) {
        console.log(o.tags)
      }
    }
  }
};
```

I'm closing the issue. Feel free to reopen if we missed something.

**Comment by sammachin** [comment]
@tatomyr Yes I realise the root level tags wern't valid but I'm not really using those anyway in my definition it was just an example.

The issue is still relating to the fact that you can't trigger a function against operation level tags, whereas you can for other items like operation params.
I'm doing it like you suggest in my code but I think it would still be better if i could act on the tags directly so IMO the bug still stands.
